src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -1085,7 +1085,8 @@ trait Types
     override def baseTypeSeq: BaseTypeSeq = supertype.baseTypeSeq
     override def baseTypeSeqDepth: Depth = supertype.baseTypeSeqDepth
     override def baseClasses: List[Symbol] = supertype.baseClasses
-  }
+    override def boundSyms: Set[Symbol] = emptySymbolSet
```

Comment:
Yep, I agree. Things seem to work with your proposed changes. I'm pushed a commit that actually avoids calling `boundSyms` altogether, which avoids the overhead of creating sets just to check if one of the `from` symbols exists in there, I'm pretty sure it will be faster to iterate over, e.g, `MethodSymbol#typeParams` directly and compare search for element to `from`.

We can actually short-circuit that search in many cases by pre-computing the range of symbol ids in `from` and by noting whether it has any term symbols. I've pushed a commit with that does just that...